lib/ansible/plugins/loader.py

```diff
@@ -818,9 +1010,48 @@ def _load_plugin_filter():
     return filters
 
 
+# since we don't want the actual collection loader understanding metadata, we'll do it in an event handler
+def _on_collection_load_handler(collection_python_package_name):
+    # parse collection metadata (if present) and set it under the package's _collection_meta
+    # FUTURE: check collection metadata's declared Ansible version support string and warn/fail on mismatch
+    metadata_package_name = collection_python_package_name + ".meta"
+    collection_meta_raw = pkgutil.get_data(metadata_package_name, 'routing.yml')
+
+    if not collection_meta_raw:
+        collection_meta_raw = '{}'
+
+    collection_package = sys.modules.get(collection_python_package_name)
+
+    collection_meta = None
+
+    try:
+        collection_meta = yaml.load(collection_meta_raw, Loader=SafeLoader)
+    except Exception as ex:
+        display.warning("Error parsing collection metadata from {0}: {1}".format(metadata_package_name, ex))
+
+    collection_package._collection_meta = collection_meta
+
+    try:
+        if not _does_collection_support_ansible_version(collection_meta.get('requires_ansible', ''), ansible_version):
```

Comment:
From the PR

>  Bikeshed location of requires_ansible key

I think this code implies that it's a top-level key in the `meta/routing.yml` file right?